Fixed: #3510: Normal user can not access '$guest_account=true' pages

ddb4github opened the pull request:
For `$guest_account=true` case only

Commit:
Fixed: #3510: Normal user can not access $guest_account' pages

netniV reviewed (commented):
Something feels off with this but i can't put my finger on it.  I think it might be because you are returning true even though $guest_user_id might not be set (since get_guest_account() could return without a guest account if none are setup).

TheWitness reviewed (changes requested):
This is what I meant when I said overly complicated.  Jing, we do need a CHANGELOG.  Otherwise, I'm fine.  @netniV, consider it approved after the CHANGELOG is sufficient.

Commit:
Update Changelog
Fixed: Ensure "$guest_account=true" pageis is avaliable only if Settings->Authentication->Guest User is configured

TheWitness commented:
Basically, what Jing is saying is that if you are logged in, and don't have access through permissions that are expressly marked as 'Public' (no login required), that without this patch you can not access them.

It really drives the point, at this stage, why do we even have a guest account.  It's so easy to setup user groups.  But we should need to maintain as I know a few people that especially render graphs using the guest account from other web site.

netniV committed:
Merge branch '1.2.x' into fix3510
